contributors/design-proposals/scheduling/scheduling-policy.md

```diff
@@ -0,0 +1,781 @@
+# Scheduling Policy
+
+_Status: Draft_
+
+_Authors: @arnaudmz, @yastij_
+
+_Reviewers: @bsalamat, @liggitt_
+
+# Objectives
+
+-  Define the concept of scheduling policies
+-  Propose their initial design and scope
+
+## Non-Goals
+
+-  How taints / tolerations work
+-  How NodeSelector works
+-  How node / pod affinity / anti-affinity rules work
+-  How several schedulers can be used within a single cluster
+-  How priority classes work
+
+# Background
+
+During real-life Kubernetes architecturing we encountered contexts where role-isolation (between administration and simple namespace usage in a multi-tenant context) could be improved. So far, no restriction is possible on toleration, priority class usage, nodeSelector, anti-affinity depending on user permissions (RBAC).
+
+Identified use-cases aim to ensure that administrators have a way to restrict users or namepace when
+-  using schedulers,
+-  placing pods on specific nodes (master roles for instance),
+-  using specific priority classes,
+-  expressing pod affinity or anti-affinity rules.
+
+Mandatory (and optionally default) values must also be enforced by scheduling policies in case of:
+-  mutli-arch (amd64, arm64) of multi-os (Linux, Windows) clusters
+-  multi-az / region / failure domain clusters
+
+# Overview
+
+Implementing SchedulingPolicy implies:
+-  Creating a new resource named **SchedulingPolicy** (schedpol)
+-  Creating an **AdmissionController** that dehaves on a deny-all-but basis
+-  Allow SchedulingPolicy to be used by pods using RoleBindings or ClusterRoleBindings
+
+# Detailed Design
+
+SchedulingPolicy resource specs are composed of several main attributes:
+-  **Required** scheduling components. These list the manadory NodeSelectors, Scheduler-Name, Anti-Affinity rules, Priority class, and Tolerations and optionally valid values that have to be provided to allow scheduling.
```

Comment:
If an admin wanted to impose defaults, they'd specify them both as defaults and as required?

Reply:
@bgrant0607 - Yes, if you want to enforce a default i.e. if not set add a default, otherwise if the field already exists with a different value then it fails to schedule.